libgeopmd/src/RuntimeStats.cpp

```diff
@@ -0,0 +1,181 @@
+
+    int RuntimeStats::num_metric(void) const
+    {
+        return m_metric_names.size();
+    }
+
+    void RuntimeStats::check_index(int metric_idx, const std::string &func, int line) const
+    {
+        if (metric_idx < 0 || (size_t)metric_idx >= m_metric_names.size()) {
+            throw Exception("RuntimeStats::" + func  + "(): metric_idx out of range: " + std::to_string(metric_idx),
+                            GEOPM_ERROR_INVALID, __FILE__, line);
+        }
+    }
+
+    std::string RuntimeStats::metric_name(int metric_idx) const
+    {
+        check_index(metric_idx, __func__, __LINE__);
+        return m_metric_names[metric_idx];
+    }
+
+    uint64_t RuntimeStats::count(int metric_idx) const
+    {
+        check_index(metric_idx, __func__, __LINE__);
+        return m_moments[metric_idx].count;
+    }
+
+    double RuntimeStats::first(int metric_idx) const
+    {
+        check_index(metric_idx, __func__, __LINE__);
+        double result = NAN;
+        if (m_moments[metric_idx].count != 0) {
+            result = m_moments[metric_idx].first;
+        }
+        return result;
+    }
+
+    double RuntimeStats::last(int metric_idx) const
+    {
+        check_index(metric_idx, __func__, __LINE__);
+        double result = NAN;
+        if (m_moments[metric_idx].count != 0) {
+            result = m_moments[metric_idx].last;
+        }
+        return result;
+    }
+
+    double RuntimeStats::min(int metric_idx) const
+    {
+        check_index(metric_idx, __func__, __LINE__);
+        double result = NAN;
+        if (m_moments[metric_idx].count != 0) {
+            result = m_moments[metric_idx].min;
+        }
+        return result;
+    }
+
+    double RuntimeStats::max(int metric_idx) const
+    {
+        check_index(metric_idx, __func__, __LINE__);
+        double result = NAN;
+        if (m_moments[metric_idx].count != 0) {
+            result = m_moments[metric_idx].max;
+        }
+        return result;
+    }
+
+    double RuntimeStats::mean(int metric_idx) const
+    {
+        check_index(metric_idx, __func__, __LINE__);
+        double result = NAN;
+        if (m_moments[metric_idx].count != 0) {
+            result = m_moments[metric_idx].m_1 /
+                     m_moments[metric_idx].count;
+        }
+        return result;
+    }
+
+    double RuntimeStats::std(int metric_idx) const
+    {
+        check_index(metric_idx, __func__, __LINE__);
+        double result = NAN;
+        if (m_moments[metric_idx].count > 1) {
+            result = std::sqrt(
+                         (m_moments[metric_idx].m_2 -
+                          m_moments[metric_idx].m_1 *
+                          m_moments[metric_idx].m_1 / m_moments[metric_idx].count) /
+                         (m_moments[metric_idx].count - 1));
+        }
+        return result;
+    }
+
+    double RuntimeStats::skew(int metric_idx) const
+    {
+        throw Exception("RuntimeStats::" + std::string(__func__) + " not yet implemented",
```

Comment:
We should probably just drop these functions from the interface until they have an implementation.